### PR TITLE
[Beta] Add deployment workflow for beta environment

### PR DESCRIPTION
**Pull request by f-fsantos**
## What are the changes?

This commit adds a new GitHub Actions workflow file, deploy-beta.yml, which is responsible for deploying the application to the beta environment.
This commits changes the deploy.yml GitHub Actions workflow file to only trigger a deploy to the production environment when a release is generated.

## Why am I doing these changes?

Adding support for the new beta environment.

## What did change?

Added deploy-beta.yml and changed deploy.yml triggers.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by OrangeRed — APPROVED** [review]
:pray:

**Review by OrangeRed — APPROVED** [review]

**Review by flx-sta — COMMENTED** [review]
Do we really need an env for every environment ? Normally u have 1 env and all the others are set inside the deployment script

File: src/utils.ts [hunk 1]
```diff
@@ -288,8 +288,7 @@ export const isLocal = (
 export const localServerUrl = import.meta.env.VITE_SERVER_URL ?? `http://${window.location.hostname}:${window.location.port+1}`;
 
 // Set the server URL based on whether it's local or not
-export const serverUrl = isLocal ? localServerUrl : "";
-export const apiUrl = isLocal ? serverUrl : "https://api.pokerogue.net";
+export const apiUrl = localServerUrl ?? "https://api.pokerogue.net";
```
Comment (flx-sta) [inline src/utils.ts:291]: Finally! Thanks for that

**Review by flx-sta — APPROVED** [review]
need this ASAP 🚀

**Review by brain-frog — APPROVED** [review]

**Pull request merged**